src/mpyl/cli/plan.py

```diff
@@ -0,0 +1,100 @@
+"""Commands related to plan"""
+
+from dataclasses import dataclass
+from pathlib import Path
+
+import click
+from rich.console import Console
+
+from . import CONFIG_PATH_HELP
+from . import create_console_logger
+from ..constants import (
+    DEFAULT_CONFIG_FILE_NAME,
+    DEFAULT_RUN_PROPERTIES_FILE_NAME,
+)
+from ..project import Stage
+from ..run_plan import discover_run_plan, RunPlan
+from ..steps.models import ConsoleProperties
+from ..utilities.pyaml_env import parse_config
+from ..validation import validate_run_properties
+
+
+@dataclass(frozen=True)
+class Context:
+    run_properties: dict
+    config: dict
+    console: Console = create_console_logger(show_path=False, max_width=0)
+
+
+@click.group("plan")
+@click.option(
```

Comment:
Currently [printing a run plan requires a list of all stages](https://github.com/Vandebron/gh-mpyl/blob/main/src/mpyl/reporting/formatting/markdown.py#L68-L72), I'm guessing so that a run plan with no tests still prints something like:
```
Build: project-a, project-b
Tests: nothing to do
Deploy: project-a
```

I'm happy to change that, but I'd rather do it in a follow-up PR.

Reply:
We discussed this offline, here's the summary:

In Click there's the notion of "groups" and "commands", and we decided to add the `--properties` and `--config` input parameters to the **group** so that they're available for all commands under that group (example: [`projects.py`](https://github.com/Vandebron/gh-mpyl/blob/d0b98f236f4d8cf220e818e6b7570cfdb06f1333/src/mpyl/cli/projects.py#L43), and for [`build.py`](https://github.com/Vandebron/gh-mpyl/blob/d0b98f236f4d8cf220e818e6b7570cfdb06f1333/src/mpyl/cli/build.py#L50)). This means we call them like:
```
mpyl projects -c config-file.yaml list
```
```
mpyl build -c config-file.yaml run --stage deploy … 
```

When I added the new group `plan` I kept the same convention and added both `--config` and `--properties` to it to the **group** for consistency. You're right that in the case of `mpyl plan print` the config is totally unnecessary, but if I make it a command parameter then the CLI will have an inconsistent order of parameters between different invocations and that's IMHO much worse:
```
mpyl projects -c config-file.yaml list
```
```
mpyl build -c config-file.yaml run --stage deploy … 
```
```
mpyl plan create -c config-file.yaml   // yikes 😿 
```

---

We decided that a better option would be to remove all parameters from all groups, and move them to the commands only when they're necessary.
This seems to be a better approach because:
* we only have parameters in the commands where we actually need them
* we no longer need to define `Context` classes just so we can pass some context between Click groups and commands
* the order of parameters within a command is no longer relevant


We'll apply this in a follow-up PR, so we don't add more to this already huge change.